account/account.go

```diff
@@ -824,3 +824,49 @@ func FmtCalldataCairo2(fnCalls []rpc.FunctionCall) []*felt.Felt {
 
 	return execCallData
 }
+
+type DeployOptions struct {
+	ClassHash           *felt.Felt
+	MaxFee              *felt.Felt
+	DeploytWaitTime     time.Duration
+	ConstructorCalldata []*felt.Felt
+}
+
+func (account *Account) DeployAccount(options DeployOptions) (*rpc.AddDeployAccountTransactionResponse, error) {
+
+	pub, err := utils.HexToFelt(account.publicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create transaction data
+	tx := rpc.DeployAccountTxn{
+		Nonce:               &felt.Zero,
+		MaxFee:              options.MaxFee,
+		Type:                rpc.TransactionType_DeployAccount,
+		Version:             rpc.TransactionV1,
+		Signature:           []*felt.Felt{},
+		ClassHash:           options.ClassHash,
+		ContractAddressSalt: pub,
+		ConstructorCalldata: options.ConstructorCalldata,
+	}
+
+	precomputedAddress, err := account.PrecomputeAddress(&felt.Zero, pub, options.ClassHash, tx.ConstructorCalldata)
+	if err != nil {
+		return nil, err
+	}
+
+	err = account.SignDeployAccountTransaction(context.Background(), &tx, precomputedAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := account.AddDeployAccountTransaction(context.Background(), tx)
```

Comment:
You will need to wrap the tx in an rpc.BroadcastDeployAccountTxn type

Reply:
Merged the upstream main branch with this PR.